Start salt-minion daemon post-install

dlindquist opened the pull request:
I noticed - the *_git_post functions start the minion daemon while the stable/daily versions do not.  

I'm interested in changing that behavior - what would be the favored way to do that?

I see a few options:
1) Start the daemon in all cases
2) Restart the daemon in all cases
3) Add an additional command line option to allow users to choose whether to start the daemon post-install

I'd be happy to contribute the change for this issue.  

Thanks!

David

akoumjian commented:
Which distro are you seeing is not starting the daemon? All of the stable I'm seeing start the daemon. Ubuntu starts it automagically, so it isn't specified. Starting the daemon should be the default behavior, but I like the idea of a flag to disable it.

dlindquist commented:
I'm using Ubuntu 12.04.  

Ahh, okay - my mistake was partially expecting it to start regardless of whether a new version of the Ubuntu package was installed.  However, if I uninstall the salt-minion package before running bootstrapping, it still doesn't start for me since config_minion() runs after install_ubuntu_stable() and the pre-allocated keys and minion config are required to successfully start the minion for me.

Moving the "# Configure Salt" block above the "# Install Salt" block seems to fix the issue.  Shall I submit a patch?

UtahDave commented:
@dlindquist yeah, propose a fix and send a pull req and we'll discuss it.  Thanks for reporting this.

dlindquist committed:
Place minion keys and config before install

https://github.com/saltstack/salt-bootstrap/issues/14

akoumjian commented:
@s0undt3ch: This looks okay to me, but I'm not sure if you can think of a reason this might break anything. I think it should always work fine to copy keys/config into place before installing salt.

s0undt3ch commented:
@akoumjian I don't think of any issues, right now.
If we do encounter we can add another config function supposed to run after the install function.

@dlindquist Since you're the author of this change, can you please also change [the comment ordering](https://github.com/dlindquist/salt-bootstrap/blob/develop/bootstrap-salt-minion.sh#L418) to your proposed order?

akoumjian commented:
@dlindquist Also, if you could possibly separate out these pull requests. Since you're pointing to your repo's head, we can't accept one without the other. Try creating separate branches and PRing that way.

Once the comments have been rearranged I'm comfortable merging this pr in.

dlindquist committed:
Change comment ordering

dlindquist committed:
Trigger config_minion for debian

dlindquist commented:
@akoumjian Of course, sorry about that.  Updated the comment ordering.  I also saw an issue with the original commit for the debian git installation and added a fix.

s0undt3ch commented:
@dlindquist why did you remove the

``` bash
if [ "$TEMP_CONFIG_DIR" = "null" ]; then
```

check? You know that like this you're ignoring what ever config dir was passed with `--config-dir`

s0undt3ch commented on bootstrap-salt-minion.sh:
```diff
@@ -596,18 +594,17 @@ install_debian_60_git_deps() {
 }
 
+config_debian_60_git_minion() {
+    TEMP_CONFIG_DIR="${SALT_GIT_CHECKOUT_DIR}/conf/"
```
Just call `config_debian_git_minion` here.

s0undt3ch commented on bootstrap-salt-minion.sh:
```diff
@@ -580,15 +579,14 @@ install_debian_git_deps() {
 }
 
+config_debian_git_minion() {
+    TEMP_CONFIG_DIR="${SALT_GIT_CHECKOUT_DIR}/conf/"
```
This needs to be

``` bash
    if [ "$TEMP_CONFIG_DIR" = "null" ]; then
        TEMP_CONFIG_DIR="${SALT_GIT_CHECKOUT_DIR}/conf/"
        CONFIG_MINION_FUNC="config_minion"
    fi
```

This way you don't ignore `--config-dir`.

This is from my head, so, it's untested, but I think this is how it should be.

s0undt3ch replied:
Oh, and this was caused because of the function calls reordering.

s0undt3ch replied:
We should also check if this re-ordering and config triggering also affects other distros.

dlindquist committed:
Fix config, review comment

dlindquist commented:
@s0undt3ch My mistake - please see added commit.  Thanks!

s0undt3ch commented on bootstrap-salt-minion.sh:
```diff
@@ -580,34 +579,33 @@ install_debian_git_deps() {
     if [ "$TEMP_CONFIG_DIR" = "null" ]; then
         TEMP_CONFIG_DIR="${SALT_GIT_CHECKOUT_DIR}/conf/"
-        CONFIG_MINION_FUNC="config_minion"
+        config_minion
```
``` sh
config_debian_git_minion() {
    if [ "$TEMP_CONFIG_DIR" = "null" ]; then
        TEMP_CONFIG_DIR="${SALT_GIT_CHECKOUT_DIR}/conf/"
    fi
    config_minion
}
```

I think this is how it should be.

s0undt3ch commented:
Also, if no `--config-dir` is passed the default function does not run.

s0undt3ch commented:
I'll merge this, but I'm going to fix it afterwards.

s0undt3ch commented:
All fixed in b324df0b4be15d32825c840bc897b36158772619

Thanks @dlindquist!

dlindquist commented:
Thank you!!

s0undt3ch commented:
Anytime!
